src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.storage.Storage;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(String fruit, int quantity) {
+        Storage.fruits.put(fruit, Storage.fruits.get(fruit) + quantity);
```

Comment:
The current implementation incorrectly adds the purchase quantity to the existing stock. For a purchase operation, you should subtract the quantity from the current stock. Consider checking if the stock is sufficient before subtracting to avoid negative values.